pkg/scheduler/queue/queue.go

```diff
@@ -21,29 +21,33 @@ var (
 	ErrStopped         = errors.New("queue is stopped")
 )
 
-// UserIndex is opaque type that allows to resume iteration over users between successive calls
+// QueueIndex is opaque type that allows to resume iteration over tenants between successive calls
 // of RequestQueue.GetNextRequestForQuerier method.
-type UserIndex struct {
-	last int
-}
+type QueueIndex int // nolint:revive
 
-// Modify index to start iteration on the same user, for which last queue was returned.
-func (ui UserIndex) ReuseLastUser() UserIndex {
-	if ui.last >= 0 {
-		return UserIndex{last: ui.last - 1}
+// Modify index to start iteration on the same tenant, for which last queue was returned.
+func (ui QueueIndex) ReuseLastIndex() QueueIndex {
+	if ui < 0 {
+		return ui
 	}
-	return ui
+	return ui - 1
 }
 
-// FirstUser returns UserIndex that starts iteration over user queues from the very first user.
-func FirstUser() UserIndex {
-	return UserIndex{last: -1}
-}
+// StartIndex is the UserIndex that starts iteration over tenant queues from the very first tenant.
+var StartIndex QueueIndex = -1
 
 // Request stored into the queue.
-type Request interface{}
+type Request any
+
+// RequestChannel is a channel that queues Requests
+type RequestChannel chan Request
+
+type Queue interface {
```

Comment:
Maybe I missed it t where's the interface used?

Reply:
It's implemented by `RequestQueue`. But I just realised that the `EnqueueRequest` function signature is wrong. Should be `Enqueue`. I will remove the interface because it's not used.